### PR TITLE
Use icons from frontend-shared

### PR DESCRIPTION
**Pull request by LMS007**
**depends on https://github.com/hypothesis/frontend-shared/pull/28**

- Update all references to use registerIcons from frontend-shared
- Update all references to use SvgIcon from frontend-shared
- Remove redundant icons "arrow-right" and "arrow-left" and used shared versions instead
- Add shared "cancel" icon
- Remove cloned SvgIcon and tests from LMS
- Fix css class formatting for .StudentSelector__change-student
- Add yalc files to gitignore--used for linking locally to shared package

-------

Precursor work needed to start updating the Buttons, specifically to use IconButton. _No visual changes_

### CODE REVIEW

File: lms/static/scripts/frontend_apps/icons.js
Comment (LMS007) [inline lms/static/scripts/frontend_apps/icons.js:16]: We will need this for creating the cancel "X" icon button later.

File: lms/static/styles/components/_StudentSelector.scss
Comment (LMS007) [inline lms/static/styles/components/_StudentSelector.scss:12]: the left/right arrow icons pulled in from the shared package are 16x16, rather than 24x24, this forces them to be 24x24

Ideally we should use a mixin utility such as 

```@mixin icon--medium {
  width: var.$icon-size--medium;
  height: var.$icon-size--medium;
}
```

But these only exist on the client at present. Its likely we may bring them into the shared package at some point, but I think hard-coding the values for now is okay.

**Comment by robertknight** [comment]
> Add shared "cancel" icon

Do you mean "use the 'cancel' icon from the shared package introduced in https://github.com/hypothesis/frontend-shared/pull/27"?

**Review by robertknight — CHANGES_REQUESTED** [review]
The code changes in this PR look fine, but CI is failing. It looks to me like this PR has a dependency on https://github.com/hypothesis/frontend-shared/pull/27? If so, please indicate that at the top of the PR description, along with a note about any changes that will be needed once the dependency is merged.

File: lms/static/styles/components/_StudentSelector.scss
Comment (lyzadanger) [reply lms/static/styles/components/_StudentSelector.scss:12]: As we move forward, icon (SVG) sizing should be handled with proportional (`em`) units, based on the contextual local `font-size`. Thus, all icons should be sized `1em` both in width and height, and the pixel size of that would be based on the `font-size`. You can see this is the case in the styling for the shared button components, for example. 

What I'd suggest for now (until we can be more systematic here) is:

* Set a `font-size` on an appropriate parent element. You can hard-code `24px` for now if there is no better way to do it. I'll leave that up to you.
* Set width/height rules on this selector to `1em`

The longer-term fix will involve updating how `SvgIcon` styling works. But we'll need an interim step.

The main goal being: SVG sizing based on `em` units based on local font context.

**Review by lyzadanger — CHANGES_REQUESTED** [review]
Hi, @LMS007 

Thanks for working on this.

Can you please break this single commit up? I'm thinking there might be three commits here (but I'll leave that to you):

* `.gitignore` changes
* Use `SvgIcon` from `frontend-shared` and remove local `SvgIcon`-related assets
* Update a few specific icons to use shared variants and remove the local icons corresponding to them

I also left a note about icon sizing...

**Comment by LMS007** [comment]
@lyzadanger I updated that svg to just use 1.5em. Its close to ~24px and just gets us out of having to drop in a px anywhere awkwardly on a parent that later needs to be ripped out too. The fact is this is going to change soonish anyway because we're replacing this with an IconButton and probably adjusting the sass vars in lms too.

**Review by lyzadanger — CHANGES_REQUESTED** [review]
Hi, @LMS007 — please do go ahead and use a `px` font-size on the parent element, and set the icon sizing to `1em`. `1.5em` is magic at this point without explicitly controlling what it's relative _to_.  It's, granted, not _likely_ that font sizing will change in an element from which this inherits anytime soon, but it's _possible_, and I don't want to leave this in a breakable state.

Also, can you please break this into multiple constituent commits? Thanks!

**Review by lyzadanger — CHANGES_REQUESTED** [review]
Hi, can you please break this down into multiple commits as suggested in the first review feedback?

**Pull request closed without merging**
